libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -542,6 +544,34 @@ class
     Specification Integer
   cardinalTypeSpec = cardinalTypeSpec @(SimpleRep a)
 
+------------------------------------------------------------------------
+-- Some instances of HasSpec
+------------------------------------------------------------------------
+
+instance HasSpec Bool where
+  type TypeSpec Bool = ()
+  emptySpec = ()
+  combineSpec _ _ = typeSpec ()
+  genFromTypeSpec _ = pureGen arbitrary
+  cardinalTypeSpec _ = equalSpec 2
+  cardinalTrueSpec = equalSpec 2
+  shrinkWithTypeSpec _ = shrink
+  conformsTo _ _ = True
+  toPreds _ _ = toPred True
```

Comment:
I think it is important to leave some comments here about how `caseOn`  will not work for Bool, and how one must use ifTrue and ifFalse instead.  Might also make some comments on `caseBoolSpec` about how that is independent of how the HasSpec for Bool is implemented.